contracts/hyphen/LiquidityPool.sol

```diff
@@ -271,6 +330,32 @@ contract LiquidityPool is
         uint256 toChainId,
         string calldata tag
     ) external payable whenNotPaused nonReentrant {
+        uint256 rewardAmount = _depositNative(receiver, toChainId);
+        emit Deposit(_msgSender(), NATIVE, receiver, toChainId, msg.value + rewardAmount, rewardAmount, tag);
+    }
+
+    function depositNativeAndSwap(
+        address receiver,
+        uint256 toChainId,
+        string calldata tag,
+        SwapRequest[] memory swapRequest
+    ) external payable whenNotPaused nonReentrant {
+
+        uint256 totalPercentage = 0;
+        for (uint256 index = 0; index < swapRequest.length; ++index) {
```

Comment:
create separate variable for swapRequest.length and wrap the whole loop in unchecked {} to save gas